app/controllers/concerns/breadcrumbed.rb

```diff
@@ -0,0 +1,22 @@
+module Breadcrumbed
+  extend ActiveSupport::Concern
+
+  def prepare_default_activity_trail(activity)
+    if activity.historic?
+      add_breadcrumb "Historic activities", historic_organisation_activities_path(activity.organisation)
+    else
+      add_breadcrumb "Current activities", organisation_activities_path(activity.organisation)
+    end
+
+    if activity.third_party_project?
+      add_breadcrumb activity.parent.parent.title, organisation_activity_financials_path(activity.parent.parent.organisation, activity.parent.parent)
+      add_breadcrumb activity.parent.title, organisation_activity_financials_path(activity.parent.organisation, activity.parent)
+    end
+
+    if activity.project?
+      add_breadcrumb activity.parent.title, organisation_activity_financials_path(activity.parent.organisation, activity.parent)
+    end
```

Comment:
Not a huge issue, but it might be clearer if we did an elsif here, because it's an either/or situation, i.e:

```ruby
if activity.project?
...
elsif activity.third_party_project?
...
end
```

Reply:
Good point, "we know" that being a project is mutually exclusive with being a third-party project, but it might not be obvious, and if someone at some point redefined "project" to mean either, it would break a bunch of stuff, including this.